Refactor activity.js and activity-works.js to Remove Redundant Code and Improve Code Quality

ravjot07 opened the pull request:
**Overview**
This pull request refactors the activity.js and activity-works.js file to eliminate redundant code, enhance readability, and improve overall code quality. By organizing the code into a DrawingApp class and removing unnecessary repetitions, I made the codebase more maintainable and accessible for current and future contributor

**Changes Made:-**

- Removed Redundant Code by Consolidating similar Functions
- Wrapped related variables and functions within the DrawingApp class to encapsulate the this activities state and behaviour. Improved the modularity of the code.
- Simplified the way event handlers are attached to objects, reducing complexity. And Ensured consistent use of this by leveraging closures and binding methods where appropriate.
- Applied consistent code formatting for better readability.

quozl commented:
* [ ] commit message does not follow our conventions; https://github.com/sugarlabs/sugar-docs/blob/master/src/contributing.md#making-commits

Your changes would have been better done in more than one commit, separating formatting changes from refactoring changes.  It would have made it easier to review.

ravjot07 commented:
@quozl i have changes commit message to a better suitable one. For review you can just test all the features i have created a drawing app class to encapsulate all the working and removed duplicate code from activity-works.js. like most of the code was redundant there for some reason...

ravjot07 commented:
@quozl can you review this pr and if there are changes you can let me know....

quozl commented:
The diff is unreadable due to length (276 + 528 lines) and change complexity, as if you have made edits while testing, then made one commit at the end.  Please break the commit up into smaller commits that show the changes in structure separate to changes in naming.

You suggest that review can reduce to testing of features.  What are the features you mention?  I've looked at README.md and run the activity and it is not clear what you mean by features.

@chimosky, you were involved in the previous pull request, have you a chance to review this one?  Thanks.

ravjot07 commented:
> The diff is unreadable due to length (276 + 528 lines) and change complexity, as if you have made edits while testing, then made one commit at the end. Please break the commit up into smaller commits that show the changes in structure separate to changes in naming.
> 
> You suggest that review can reduce to testing of features. What are the features you mention? I've looked at README.md and run the activity and it is not clear what you mean by features.
> 
> @chimosky, you were involved in the previous pull request, have you a chance to review this one? Thanks.

@quozl I meant working of this activity by features like drawing, numbers swapning, and clearing canvas ets.. sure i will break this PR into smaller commits and add one more PR with these changes....

quozl commented:
Thank you.  Looking forward to your changes.  You can and should replace the commit on this branch with a series of commits, rather than create a new pull request.  A new pull request displaces the discussion and confuses later readers.  But if you are lacking in time to do this, then a link back to the old pull request is at least a minimum courtesy.

ravjot07 commented:
@quozl I had updated it with my local branch in which i refactored the code can you review this one. As merging this PR will make this code much more readable for contributors....

chimosky commented on js/activity-works.js:
```diff
@@ -286,4 +286,4 @@ define(function (require) {
         }
     });
-});
+});
```
I don't know how you keep removing the new line at EOF, but you can add it with `echo >> filename`.

ravjot07 replied:
sure @chimosky i will use echo form now on

quozl replied:
Using echo is fixing it after the event.  A useful habit is to
* have a view of `git diff` open so you can see what you have actually changed; with that you should have spotted the removal of newline,
* use an interactive `git add` so you can _approve_ what to add to each commit; with that you should have avoided inserting the removal of newline into a commit, and we wouldn't have to see it.

There are software tools and plugins that provide these techniques.  I use Emacs and Magit to do it, as well as a Terminal.  I've heard others using VScode, but I don't know the plugin names.
